FinSearchUnified/Bundle/ProductNumberSearch.php

```diff
@@ -286,10 +286,14 @@ private function fetchSelectedFilterByUserCondition(
 
         $condition = $criteria->getUserCondition($facetName);
 
-        return $this->createSelectedFilter(
-            $criteriaFacet,
-            $condition
-        );
+        if ($condition) {
```

Comment:
Try to apply [Defensive Coding](https://en.wikipedia.org/wiki/Defensive_programming) in this case. I'd suggest returning null only if there is no condition.

```php
if (!$condition) {
    return null;
}

// Do the usual stuff.
```

Reply:
Maybe it makes sense to point out in a comment, when a condition may not be able to be fetched.

Reply:
Isn't it clear, when the line above is trying to get a `UserCondition`?

Reply:
That's why I said "maybe". Judge yourself if it is necessary :).